internal/ociutils/ociutils.go

```diff
@@ -0,0 +1,240 @@
+package ociutils
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"path/filepath"
+
+	"github.com/opencontainers/go-digest"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/rancher/fleet/internal/manifest"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content/memory"
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+	"oras.land/oras-go/v2/registry/remote/retry"
+)
+
+const (
+	fileType     = "application/fleet.file"
+	artifactType = "application/fleet.manifest"
+
+	OCISecretUsername  = "username"
+	OCISecretPassword  = "password"
+	OCISecretURL       = "url"
+	OCISecretBasicHTTP = "http"
+	OCISecretInsecure  = "insecure"
+)
+
+var (
+	orasStoreNew     = newOrasStore
```

Comment:
As discussed offline and in the context of another pull request using a similar approach, it would be great if we could avoid hacks like this for the sake of testing. Would integration tests be the way to go here?

Reply:
As discussed offline I've changed this to a different approach, using Interface/Struct using the interface approach. 
I'm afraid it is a bit harder to catch some of the errors from integration tests (specially the error case branches)

Reply:
I like this new approach much more, thanks for the great work!